src/passes/pass.cpp

```diff
@@ -192,7 +193,7 @@ void PassRunner::run() {
       std::cerr << "final module does not validate\n";
       abort();
     }
-    passDebug = currPassDebug;
+    recursion--;
```

Comment:
Actually I'm refactored this now, I think the concept of a nested pass runner is an important one, so I'm convinced to go more in the direction you suggested. I propose we have a `NestedPassRunner` class that handles this, instead of a method (see commit), since I think we'll need the fact it's nested in more places. Thoughts?